adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java

```diff
@@ -258,6 +259,17 @@ public ResourceIdentifier topic() {
         return topic;
     }
 
+    /**
+     * Apply a mapped targetAddress to this mqtt context.
+     * <p>
+     * This overwrites the topic determined from the message.
```

Comment:
How about: "Tenant and device identifiers included in the address override corresponding values derived from the message topic."?

Reply:
Corresponding to my other comment about the `tenant()` method the tenant id can be omitted here:
 "The device identifier included in the address overrides the device identifier derived from the message topic."